### PR TITLE
Update documentation for plot_graph.

### PR DESCRIPTION
**Pull request by indhub**
@mli, @zackchase, @madjam, @nswamy, @Roshrini, @jiajiechen
Please review and merge.

### CODE REVIEW

File: python/mxnet/visualization.py [hunk 1]
```diff
@@ -166,31 +166,44 @@ def print_layer_summary(node, out_shape):
...
+        A Graphviz digraph object visualizing the computation graph to compute `symbol`.
+
+    Example
+    -------
+    >>> net = mx.sym.Variable('data')
+    >>> net = mx.sym.FullyConnected(data=net, name='fc1', num_hidden=128)
+    >>> net = mx.sym.Activation(data=net, name='relu1', act_type="relu")
+    >>> net = mx.sym.FullyConnected(data=net, name='fc2', num_hidden=10)
+    >>> net = mx.sym.SoftmaxOutput(data=net, name='out')
+    >>> mx.viz.plot_network(net, shape={'data':(100,200)}, node_attrs={"fixedsize":"false"})
```
Comment (piiswrong) [inline python/mxnet/visualization.py]: explain how to used the returned object

File: python/mxnet/visualization.py [hunk 2]
```diff
@@ -166,31 +166,44 @@ def print_layer_summary(node, out_shape):
...
+        Specifies the shape of the input tensors. If specified, the visualization will include
+        the shape of the tensors between the nodes. `shape` is a dictionary mapping
+        input symbol names (str) to the corresponding tensor shape (tuple).
+    node_attrs: dict, optional
+        Specifies the attributes for nodes in the generated visualization. `node_attrs` is
+        a dictionary of Graphviz attribute names and values. For example,
+            ``node_attrs={"shape":"oval","fixedsize":"false"}``
+            will use oval shape for nodes and allow variable sized nodes in the visualization.
+    hide_weights: bool, optional
+        Whether to hide inputs with names of the form `*_weight` in the visualization.
```
Comment (piiswrong) [inline python/mxnet/visualization.py]: weight or bias.
also explain why they are hidden

File: python/mxnet/visualization.py [hunk 3]
```diff
@@ -166,31 +166,44 @@ def print_layer_summary(node, out_shape):
...
-        for example:
-            node_attrs={"shape":"oval","fixedsize":"fasle"}
-            means to plot the network in "oval"
-    hide_weights: bool
-        If True (default), then inputs with names like `*_weight`
-        or `*_bias` will be hidden.
+        A symbol from the computation graph. The generated digraph will visualize the part
+        of the computation graph required to compute `symbol`.
+    shape: dict, optional
+        Specifies the shape of the input tensors. If specified, the visualization will include
```
Comment (piiswrong) [inline python/mxnet/visualization.py]: the new description seems unnecessarily verbose and even harder to understand.

Comment (indhub) [inline python/mxnet/visualization.py]: If you have specific suggestions on how to rewrite any of them, please let me know.

Comment (zackchase) [inline python/mxnet/visualization.py]: The new description appears to be inaccurate. *_weight and *_bias ARE required to compute the symbol. I'd revert this back to the original description with minor modification:

If True (default), then inputs with names of form `*_weight` (corresponding to weight tensors) or `*_bias` (corresponding to bias vectors) will be hidden.

Comment (indhub) [inline python/mxnet/visualization.py]: Not sure if I understand you. Does the current description say that *_weight and *_bias are not required to compute the symbol?

**Pull request merged**